pkg/apis/core/types_shoot.go

```diff
@@ -497,6 +497,37 @@ type KubeAPIServerConfig struct {
 	EnableAnonymousAuthentication *bool
 	// EventTTL controls the amount of time to retain events.
 	EventTTL *metav1.Duration
+	// AccessControl provides authorization mechanisms for Shoot-API-Server.
+	AccessControl *AccessControl
+}
+
+// AccessControl provides authorization mechanisms for Shoot-API-Server.
+// Note: The schema (incl. child structs) and documentation resembles istio's AuthorizationPolicy, but is heavily simplified.
+type AccessControl struct {
+	// The action to take on the source of request.
+	Action *AuthorizationAction
+	// Origin of request to run defined authorization action against.
+	Source *Source
+}
+
+// AuthorizationAction is the operation (e.g. DENY) to apply on requests.
+type AuthorizationAction string
+
+const (
+	// Allows the request from a source.
+	AuthorizationAction_ALLOW AuthorizationAction = "ALLOW"
+	// Denies the request from a source.
+	AuthorizationAction_DENY AuthorizationAction = "DENY"
+)
+
+// Source holds the origin of request to which access control is applied to.
+type Source struct {
+	// A list of IP blocks, populated from the source address of the IP packet.
+	// Single IP (e.g. "1.2.3.4") and CIDR (e.g. "1.2.3.0/24") are supported.
+	IPBlocks []string
+	// A list of IP blocks, populated from X-Forwarded-For header or proxy protocol.
+	// Single IP (e.g. "1.2.3.4") and CIDR (e.g. "1.2.3.0/24") are supported.
+	RemoteIPBlocks []string
```

Comment:
Is there an expectation that `IPBlocks` and `RemoteIPBlocks` should contain similar addresses?
Does it make sense to populate just one of them?

Reply:
I also think that for the sake of simplicity there should be only one option to specify the source addresses. Depending on the way istio is deployed, configuration of istio must put these values into the required option then

Reply:
Agree, there is no added value in offering two fields (`IPBlocks` and `RemoteIPBlocks`) to the outer facing structs. I have removed `RemoteIPBlocks`. Though, as per my understanding, I now need to check if I get access to the `externalTrafficPolicy`of the deployed istio ingress in order to perform a check whether to populate istio's `IPBlocks` or `RemoteIPBlocks`. Looking into this